test/integration/certificates/trigger_controller_test.go

```diff
@@ -217,26 +200,132 @@ func TestTriggerController_RenewNearExpiry(t *testing.T) {
+	if err != nil {
+		t.Fatal(err)
+	}
+	apiutil.SetCertificateCondition(cert, 1, cmapi.CertificateConditionIssuing, cmmeta.ConditionFalse, "", "")
+	cert.Status.IssuanceAttempts = &issuanceAttempts
+	cert.Status.LastFailureTime = &metaNow
 	cert, err = cmCl.CertmanagerV1().Certificates(namespace).UpdateStatus(ctx, cert, metav1.UpdateOptions{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	err = wait.PollImmediateUntil(time.Millisecond*200, func() (done bool, err error) {
-		c, err := cmCl.CertmanagerV1().Certificates(cert.Namespace).Get(ctx, cert.Name, metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
-		if apiutil.CertificateHasCondition(c, cmapi.CertificateCondition{
-			Type:   cmapi.CertificateConditionIssuing,
-			Status: cmmeta.ConditionTrue,
-		}) {
-			return true, nil
-		}
-		return false, nil
-	}, ctx.Done())
+
+	// 2. Test that issuance is not attempted whilst in backoff period
+	// modify some cert field to ensure a reconcile gets triggered
+	t.Log("Advance clock to slightly before the end of the backoff period")
+	fakeClock.SetTime(now.Add(backoffPeriod - time.Minute))
+	someRenewalTime := metav1.NewTime(time.Now())
+	cert.Status.RenewalTime = &someRenewalTime
+	_, err = cmCl.CertmanagerV1().Certificates(namespace).UpdateStatus(ctx, cert, metav1.UpdateOptions{})
 	if err != nil {
-		t.Error("Failed waiting for Certificate to have Issuing condition")
+		t.Fatal(err)
 	}
+	t.Log("Ensuring Certificate does not have Issuing condition set to true for 2s...")
+	ensureCertificateDoesNotHaveIssuingCondition(ctx, t, cmCl, namespace, certName)
+
+	// 3. Test that issuance gets retried once the backoff period is over
+	t.Log("Advance clock to just after the backoff period")
+	fakeClock.SetTime(now.Add(backoffPeriod + time.Second))
+	// We need to update some field to trigger a reconcile (as advancing the
+	// clock does not cause the cert, that was set to be reconciled after the
+	// backoff period, to be reconciled now)
+	cert, err = cmCl.CertmanagerV1().Certificates(namespace).Get(ctx, certName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	someRenewalTime = metav1.NewTime(time.Now())
+	cert.Status.RenewalTime = &someRenewalTime
+	_, err = cmCl.CertmanagerV1().Certificates(namespace).UpdateStatus(ctx, cert, metav1.UpdateOptions{})
+	if err != nil {
+		// This is hacky, but it appears that sometimes the trigger
+		// controller gets triggered before renewal time is updated (but
+		// after the backoff time was set to have passed) and the
+		// issuing condition is already applied so this update will
+		// fail. This can be removed once we have server side apply.
+		t.Log("Ensuring Certificate does get the Issuing condition set to true after the backoff period")
+		ensureCertificateHasIssuingCondition(ctx, t, cmCl, namespace, certName)
+	}
+
+	t.Log("Ensuring Certificate does get the Issuing condition set to true after the backoff period")
+	ensureCertificateHasIssuingCondition(ctx, t, cmCl, namespace, certName)
```

Comment:
Can't quite tell what is going on here, and seems wrong that we just ignore the update status error in favour of calling `ensureCertificateHasIssuingCondition` twice...

Reply:
Thanks Josh, I've fixed this with patching instead of applying the test condition